app/main.py

```diff
@@ -1,2 +1,90 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list[None | tuple[Hashable, hash, Any]] = [None] * 8
+        self.load_factor = 2 / 3
+        self.capacity = 2
+
+    def __len__(self) -> int:
+        return self.length
+
+    @staticmethod
+    def add_item_to_table(
+            table: list[None | tuple[Hashable, hash, Any]],
+            key: Hashable,
+            value: Any
+    ) -> None:
+        hash_ = hash(key)
+        index = hash_ % (len(table) - 1)
+        while True:
+            if not table[index] or table[index][0] == key:
+                table[index] = key, hash_, value
+                break
+            index += 1
+            if index == len(table):
+                index = 0
+
+    def resize(self) -> None:
+        new_hash_table = [None] * len(self.hash_table) * self.capacity
```

Comment:
The `resize` method should rehash the keys when transferring them to the new hash table. Currently, the keys are being inserted without rehashing, which can lead to incorrect placement and potential collisions.